src/main/java/me/jellysquid/mods/sodium/client/render/vertex/VertexConsumerTracker.java

```diff
@@ -0,0 +1,16 @@
+package me.jellysquid.mods.sodium.client.render.vertex;
+
+import it.unimi.dsi.fastutil.objects.ReferenceOpenHashSet;
+import it.unimi.dsi.fastutil.objects.ReferenceSet;
+import me.jellysquid.mods.sodium.client.SodiumClientMod;
+import net.minecraft.client.render.VertexConsumer;
+
+public class VertexConsumerTracker {
+    private static final ReferenceSet<Class<? extends VertexConsumer>> BAD_CONSUMERS = new ReferenceOpenHashSet<>();
+
+    public static void logBadConsumer(VertexConsumer consumer) {
+        if(BAD_CONSUMERS.add(consumer.getClass())) {
```

Comment:
Are there thread safety concerns here? If so then a `Collections.newSetFromMap(new ConcurrentHashMap<>())` can be used.